src/avram/query_builder.cr

```diff
@@ -112,8 +114,16 @@ class Avram::QueryBuilder
     self
   end
 
-  private def distinct?
-    @distinct || @distinct_on
+  def distinct?
+    @distinct || has_distinct_on?
+  end
+
+  def has_distinct_on?
+    !!@distinct_on
+  end
+
+  def raw_distinct_on
```

Comment:
I wasn't totally sold on the name. We could just make it a property, or add a getter, but that wouldn't really be consistent with the rest of the properties. Maybe we leave this as is, and then next year do a huge cleanup refactor 🤔

Reply:
I’m not sure I get what is different between `raw_disrinct_on` and @distinct_on. Could you explain that?

Reply:
I can access `raw_distinct_on` from a separate object.  This is so I don't do something like`other_query.@distinct_on`. I could just make a `getter` for that and call `other_query.distinct_on` if that works better.